website/static/js/pages/profile-settings-addons-page.js

```diff
@@ -101,3 +101,19 @@ $('.addon-oauth').each(function(index, elem) {
     ko.applyBindings(viewModel, elem);
     viewModel.updateAccounts();
 });
+
+// Reusable function to fix affix widths to columns.
+function fixAffixWidth() {
+    $('.affix, .affix-top, .affix-bottom').each(function (){
+        var el = $(this);
+        var colsize = el.parent('.affix-parent').width();
+        el.outerWidth(colsize);
+    });
+}
+
+$(document).ready(function () {
+
+    $(window).resize(function (){ fixAffixWidth(); });
+    $('#affix-nav').on('affixed.bs.affix', function(){ fixAffixWidth(); });
+
+  });
```

Comment:
@GageGaskins For instance to refactor you can combine all the above code into a single function without document.ready and your page js file can only include 
```js
$(document).ready(function () {
      initializeResponsiveAffix();
///...
}
```